service/src/main/java/uk/gov/hmcts/reform/testingsupport/controllers/TestingSupportController.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.hmcts.reform.testingsupport.controllers;
+
+import feign.FeignException;
+import io.swagger.annotations.Api;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.fpl.enums.State;
+import uk.gov.hmcts.reform.fpl.service.ccd.CoreCaseDataService;
+import uk.gov.hmcts.reform.testingsupport.service.TestingSupportService;
+
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+import static uk.gov.hmcts.reform.fpl.enums.State.PREPARE_FOR_HEARING;
+
+@Api
+@Slf4j
+@RestController
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@ConditionalOnProperty(prefix = "testing_support", name = "enabled", havingValue = "true")
+@SuppressWarnings("unchecked")
+public class TestingSupportController {
+    private static final String POPULATE_EVENT_ID_TEMPLATE = "populateCase-%s";
+    private final CoreCaseDataService coreCaseDataService;
+    private final TestingSupportService testingSupportService;
+
+    @PostMapping("/testingSupport/populateCase/{caseId}")
```

Comment:
should be: testing-support/{caseId}/populate
testing-support as this is what is used across other teams (see https://idam-api.aat.platform.hmcts.net/swagger-ui.html#/Testing_Support)

In general rest endpoint should identify resource and should contain nouns. action should be specifiec by http method or content type. If we want verb in url should be at the end (Define resource first, then action on it)